analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -23,15 +23,17 @@ namespace SonarAnalyzer.Rules.CSharp
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class EqualityOnFloatingPoint : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S1244";
-        private const string MessageFormat = "Do not check floating point {0} with exact values, use a range instead.";
+        private const string DiagnosticId = "S1244";
+        private const string MessageFormat = "Do not check floating point {0} with exact values, use {1} instead.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+        private static readonly string[] EqualityOperators = new[] { "op_Equality", "op_Inequality" };
```

Comment:
Small simplification in case Martin's suggestion doesn't work. The implementation checks for `MethodKind.UserDefinedOperator` that I'm not sure how it behaves
```suggestion
        private static readonly string[] EqualityOperators = { "op_Equality", "op_Inequality" };
```

Reply:
Not needed, as `EqualityOperators` has been removed.